### PR TITLE
Delete profile

### PR DESCRIPTION
**Pull request by DrColver**
Replaces PR https://github.com/bisq-network/bisq2/pull/1192

Based on PR https://github.com/bisq-network/bisq2/pull/1198

Fix https://github.com/bisq-network/bisq2/issues/794

### CODE REVIEW

File: chat/src/main/java/bisq/chat/ChatService.java [hunk 1]
```diff
@@ -205,7 +207,22 @@ public Optional<TwoPartyPrivateChatChannel> createAndSelectTwoPartyPrivateChatCh
             optionalChannel.ifPresent(channel -> getChatChannelSelectionService(chatChannelDomain).selectChannel(channel));
             return optionalChannel;
         }
+    }
+
+    public boolean isIdentityUsed(UserIdentity userIdentity) {
```
Comment (HenrikJannsen) [inline chat/src/main/java/bisq/chat/ChatService.java:212]: Maybe better `isUserIdentityUsed`?

File: desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java [hunk 2]
```diff
@@ -110,11 +114,12 @@ public UserIdentity fromString(String string) {
         saveButton = new Button(Res.get("action.save"));
         saveButton.setDefaultButton(true);
 
-        deletedButton = new Button(Res.get("user.userProfile.deleteProfile"));
+        deleteButton = new Button(Res.get("user.userProfile.deleteProfile"));
+        deleteWrapper = new SplitPane(deleteButton);
```
Comment (HenrikJannsen) [inline desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java:118]: Why is a SplitPane used here?

File: user/src/main/java/bisq/user/identity/UserIdentityService.java [hunk 3]
```diff
@@ -215,14 +215,11 @@ public CompletableFuture<DataService.BroadCastDataResult> editUserProfile(UserId
                 .thenCompose(result -> networkService.publishAuthenticatedData(newUserProfile, oldIdentity.getNodeIdAndKeyPair().getKeyPair()));
     }
 
-    public CompletableFuture<DataService.BroadCastDataResult> deleteUserProfile(UserIdentity userIdentity) {
-        //todo add more checks if deleting profile is permitted (e.g. not used in trades, PM,...)
-        if (getUserIdentities().size() <= 1) {
```
Comment (HenrikJannsen) [inline user/src/main/java/bisq/user/identity/UserIdentityService.java:220]: Maybe we should keep that check? The app would be in an unexpected/untested state without any UserIdentity. I know the callers check, but that module does not know about that. The rule that at least one identity must exist is inside its domain so I think we should have the check here.

File: bisq_easy/src/main/java/bisq/bisq_easy/BisqEasyService.java [hunk 4]
```diff
@@ -105,4 +111,16 @@ public CompletableFuture<Boolean> shutdown() {
...
     }
+
+    public boolean isDeleteUserIdentityProhibited(UserIdentity userIdentity) {
+        return chatService.isIdentityUsed(userIdentity) ||
+                !userIdentityService.hasMultipleUserIdentities();
+    }
+
+    public CompletableFuture<DataService.BroadCastDataResult> deleteUserIdentity(UserIdentity userIdentity) {
+        if (isDeleteUserIdentityProhibited(userIdentity)) {
+            return CompletableFuture.failedFuture(new RuntimeException(Res.get("user.userProfile.deleteProfile.cannotDelete")));
```
Comment (HenrikJannsen) [inline bisq_easy/src/main/java/bisq/bisq_easy/BisqEasyService.java]: I think we should not reuse a UI text here as error message. Error messages are so far mostly in plain English and not from Resources and should only be escalated to the user by the error popup which can be filled with more technical data. As we check anyway at the caller that error should never be thrown anyway (but good to leave it at the service level).

**Review by HenrikJannsen — CHANGES_REQUESTED** [review]
Some nits...

File: chat/src/main/java/bisq/chat/ChatService.java (hunk 1 above)
Comment (DrColver) [reply chat/src/main/java/bisq/chat/ChatService.java:212]: I think that would be less readable considering the argument is UserIdentity.

File: desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java (hunk 2 above)
Comment (DrColver) [reply desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java:118]: A Control is needed to add the Tooltip. When the button is disabled it doesn't receive mouse events so it wouldn't show the Tooltip. Other Controls would work as well as wrappers.

File: user/src/main/java/bisq/user/identity/UserIdentityService.java (hunk 3 above)
Comment (DrColver) [reply user/src/main/java/bisq/user/identity/UserIdentityService.java:220]: Good point. I added it again.

File: bisq_easy/src/main/java/bisq/bisq_easy/BisqEasyService.java (hunk 4 above)
Comment (DrColver) [reply bisq_easy/src/main/java/bisq/bisq_easy/BisqEasyService.java]: Fixed.

File: chat/src/main/java/bisq/chat/ChatService.java (hunk 1 above)
Comment (HenrikJannsen) [reply chat/src/main/java/bisq/chat/ChatService.java:212]: Ok

File: desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java (hunk 2 above)
Comment (HenrikJannsen) [reply desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java:118]: Yes, I was just wondering why Splitpane was selected. I guess Pane would be the cheapest one as it does not support auto-layout (if that is not needed as I assume so).

**Review by HenrikJannsen — APPROVED** [review]
utACK

File: desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java (hunk 2 above)
Comment (DrColver) [reply desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java:118]: A `Control` is needed for the tooltip support. Maybe there is a better choice than SplitPane but from what I've seen it's reasonable. It's meant to take Nodes as argument and doesn't decorate it when it's just a single node. Maybe ScrollPane would be better, not sure, don't think it makes a big difference.
